File: d0ubleho/2/18310.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int N; // 집의 수
    cin >> N;

    vector<int> house(N);

    for (int i = 0; i < N; i++) {
        cin >> house[i];
    }

    // 집들의 위치를 오름차순으로 정렬
    sort(house.begin(), house.end());

    // 집의 수가 짝수면, N/2 - 1 번째 위치의 값이 중앙값에 해당(여러 개의 값이 도출될 경우 가장 작은 값을 출력)
    // 집의 수가 홀수면, N/2 번째 위치의 값이 중앙값에 해당
    cout << house[(N - 1) / 2];

    return 0;
}

File: chaekeun/2/18310.py
from sys import stdin
input = stdin.readline

n = int(input())
a = list(map(int, input().split()))
ans, minDist = 0, 200000

# # 각각의 집에 안테나를 설치해보았을 때 모든 집까지의 거리를 계산해보고 min갱신
# # 으로 하려고 했으나 시간초과..ㅎ
# for house in a:
#     sum = 0
#     # 절댓값함수 abs를 이용해서 거리를 담은 list만들고 sum함수 적용
#     sumDist = sum([abs(elem-house) for elem in a])
    
#     if sumDist < minDist:
#         ans = house
#         minDist = sumDist

# print(ans)

a.sort()
print(a[(n-1)//2])